integrations/azure-devops/main.py

```diff
@@ -0,0 +1,114 @@
+from typing import Any
+from loguru import logger
+from port_ocean.context.ocean import ocean
+from azure_devops.client import AzureDevopsClient
+from azure_devops.webhooks.webhook_event import WebhookEvent
+from port_ocean.core.ocean_types import ASYNC_GENERATOR_RESYNC_TYPE
+from azure_devops.search_criteria import (
+    PULL_REQUEST_SEARCH_CRITERIA,
+    WORK_ITEMS_WIQL_QUERY,
+    MAX_WORK_ITEMS_PER_QUERY,
+)
+from azure_devops.utils import Kind
+from bootstrap import setup_listeners, webhook_event_handler
+from starlette.requests import Request
+
+
+@ocean.on_start()
+async def setup_webhooks() -> None:
+    if ocean.event_listener_type == "ONCE":
+        logger.info("Skipping webhook creation because the event listener is ONCE")
+        return
+    if not ocean.integration_config.get("app_host"):
+        logger.warning("No app host provided, skipping webhook creation.")
+        return
+
+    azure_devops_client = AzureDevopsClient.create_from_ocean_config()
+    await setup_listeners(ocean.integration_config["app_host"], azure_devops_client)
+
+
+@ocean.on_resync(Kind.PROJECT)
+async def resync_projects(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    azure_devops_client = AzureDevopsClient.create_from_ocean_config()
+    async for projects in azure_devops_client.generate_projects():
+        logger.info(f"Resyncing projects: {str(projects)}")
+        yield projects
+
+
+@ocean.on_resync(Kind.TEAM)
+async def resync_teams(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    azure_devops_client = AzureDevopsClient.create_from_ocean_config()
+    async for teams in azure_devops_client.generate_teams():
+        logger.info(f"Resyncing teams: {str(teams)}")
+        yield teams
+
+
+@ocean.on_resync(Kind.MEMBER)
+async def resync_members(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    azure_devops_client = AzureDevopsClient.create_from_ocean_config()
+    async for members in azure_devops_client.generate_members():
+        logger.info(f"Resyncing members: {str(members)}")
+        yield members
```

Comment:
R we sure we want to export these?
they r preforming quite alot of requests

Reply:
Agreed. If not required business-wise we should remove this.